### DISCUSSION TITLE
Can I create predictable URLs for review apps?

### DISCUSSION THREAD

**Discussion by imjared** [vercel/community#468] [Help] [answered]
As our app deploys, we see that the generated domains are random strings like `mg-app-a92lqqv9b`, `mg-app-amm50cbzm`, etc. These are generated per-push to a PR.

For our CI/CD review app pipeline, it'd be nice to be able to predict the URL of a preview deployment. For example, `{PR_BRANCH_NAME}-{MY_TEAM_ID}.vercel.app`

We'd like to be able to point our backend + nginx proxy at a frontend PR domain and know that domain will not change for the lifecycle of the PR but that any new changes to the PR will be deployed at the URL we've specified.

Desired output: A PR named `update-table-header` would always have the preview URL `update-table-header-my_team_id.vercel.app`

I've seen docs about [preview deployment suffixes](https://vercel.com/docs/concepts/deployments/automatic-urls#preview-deployment-suffix) but have been drawing blanks about the prefix.

My use case seems like it's not quite using the platform as expected so I'm wondering if this is doable or if we'd run into issues if a domain already existed?

Related, maybe:
- https://github.com/vercel/community/discussions/439

**Comment by swarnava** [comment, marked as answer]
Hey,

You can rely on `VERCEL_URL` system variable which will always return URL of the deployment. Example: `my-site-7q03y4pi5.vercel.app`. Learn more here: https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables

Does  it help?

**Reply by imjared** [reply]
Hi @swarnava, thanks for the reply. Unfortunately this doesn't seem to answer the question. I'm looking for predictable URLs for a Vercel deploy that I could use in a completely different system. Our backend is on Heroku and doesn't have access to Vercel's environment variables.

Heroku's implementation for [predictable URLs for review apps](https://devcenter.heroku.com/articles/github-integration-review-apps#selecting-the-url-pattern) is almost exactly what I'd be trying to replicate.

**Reply by Pieparker** [reply]
Hey @imjared, [this has been covered before](https://github.com/vercel/community/discussions/114#discussioncomment-1897657)!

There's plenty of different options and, hopefully, there's an option here that works for you.
